Sources/Workspace/DefaultPluginScriptRunner.swift

```diff
@@ -576,7 +576,7 @@ fileprivate extension FileHandle {
     func writePluginMessage(_ message: HostToPluginMessage) throws {
         // Encode the message as JSON.
         let payload = try JSONEncoder().encode(message)
```

Comment:
unrelated to this PR, but should the JSONEncoder be shared across call instead of created every time?

Reply:
It's pretty lightweight, but it is a class, so yes, it would be better to share it as it's at least one heap allocation.  From the implementation it doesn't look as if state can bleed from one encode to the next, which was my main concern.  I'll make a note to address this in another PR.

Reply:
Also, for context, the number of messages for a typical plugin is anywhere from two to half a dozen, a bit more if there are diagnostics or a lot of build commands defined.  So it would matter but probably not measurably.  Should still be fixed.  Thanks for pointing it out.